From 914298266f6116e26672b27bdc16b083a46996bc Mon Sep 17 00:00:00 2001
From: Eric Ernst <eric.ernst@intel.com>
Date: Thu, 28 Aug 2014 17:11:20 -0700
Subject: [PATCH 1/1] EM: Update battery curve data for ECS

Currently the charging driver for PMIC makes use of
hardcoded values.

This is very unfortunate, and should be migrated to use ACPI
ASAP, most likely via OEM table in the near future.

In the meantime, we are updating the fuel gauge battery curve
and design_cap values to reflect the POR battery used
within the ECS platform (it was set for MRD-7 platform prior).

These are minor changes, and shouldn't have very strong impact
on MRD7.

Signed-off-by: Eric Ernst <eric.ernst@intel.com>
---
 drivers/mfd/intel_soc_pmic_dc.c |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index 5238746bdf16..e7eb09c727fe 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -361,7 +361,7 @@ static void platform_set_battery_data(struct dollarcove_fg_pdata *pdata,
 		pdata->design_max_volt = prof->voltage_max ? prof->voltage_max
 			: 4350;
 	} else {
-		pdata->design_cap = 4980;
+		pdata->design_cap = 4045;
 		pdata->design_max_volt = 4350;
 	}
 }
@@ -375,7 +375,7 @@ static void platform_init_chrg_params(struct dollarcove_chrg_pdata *pdata)
 static void platform_set_battery_data(struct dollarcove_fg_pdata *pdata,
 	struct ps_batt_chg_prof *chg_prof)
 {
-	pdata->design_cap = 4980;
+	pdata->design_cap = 4045;
 	pdata->design_max_volt = 4350;
 }
 
@@ -405,10 +405,10 @@ static void dc_xpwr_chrg_pdata(void)
 }
 
 static int fg_bat_curve[] = {
-	0x0, 0x0, 0x0, 0x0, 0x1, 0x1, 0x2, 0x2,
-	0x2, 0x3, 0x3, 0x4, 0xc, 0x10, 0x16, 0x1c,
-	0x27, 0x2c, 0x30, 0x35, 0x3a, 0x3f, 0x43, 0x47,
-	0x4b, 0x4e, 0x50, 0x51, 0x54, 0x57, 0x5b, 0x5e,
+	0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x2,
+	0x2, 0x3, 0x5, 0x9, 0xf, 0x18, 0x24, 0x29,
+    0x2e, 0x32, 0x35, 0x3b, 0x40, 0x45, 0x49, 0x4c,
+    0x50, 0x53, 0x55, 0x57, 0x5a, 0x5d, 0x61, 0x64,
 };
 
 static void dc_xpwr_fg_pdata(void)
-- 
1.7.9.5

